api/genericresource/parse.go

```diff
@@ -0,0 +1,35 @@
+package genericresource
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/docker/swarmkit/api"
+)
+
+// Parse parses the GenericResource resources given by the arguments
+func Parse(tprs string, resources *api.Resources) error {
+	resources.Generic = make(map[string]*api.GenericResource)
+	mtpr := resources.Generic
+
+	for _, term := range strings.Split(tprs, ";") {
```

Comment:
I think the description needs to explicitly specify what type of resource it is (discrete value, set, scalar, etc). And allow extension.